File: problem_template.py
from typing import List, Optional, Tuple, Dict, Set
from collections import defaultdict, deque, Counter, OrderedDict
from dataclasses import dataclass
import heapq
import bisect
import functools
import itertools
import math
import sys


class Solution:
    def solve(self, nums: List[int]) -> int:
        pass


@dataclass
class TestCase:
    nums: List[int]
    expected: int


if __name__ == "__main__":
    s = Solution()

    tests = [
        TestCase([10, 30, 21], 32),
        TestCase([100, 70], 92),
        TestCase([7, 3, 9], 3),
    ]

    correct = 0
    for i, t in enumerate(tests):
        actual = s.solve(t.nums)
        passed = actual == t.expected
        correct += passed

        print(f"Test {i}: {'PASS' if passed else 'FAIL'}")
        print(f"\tExpected: {t.expected}")
        print(f"\tActual:   {actual}")

    print(f"\n{correct} / {len(tests)} correct.")

File: problem_template.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int solve(std::vector<int>& nums) {
        return -1; // TODO
    }
};

struct TestCase {
    std::vector<int> nums;
    int expected;
};

int main() {
    std::vector<TestCase> tests = {
        {{10, 30, 21}, 32},
        {{100, 70}, 92},
        {{7, 3, 9}, 3}
    };

    Solution sol;
    int correct = 0;

    for (size_t i = 0; i < tests.size(); i++) {
        TestCase& t = tests[i];
        int actual = sol.solve(t.nums);
        bool passed = (actual == t.expected);
        correct += passed;

        std::cout << "Test " << i << ": " << (passed ? "PASS" : "FAIL") << "\n";
        std::cout << "\tExpected: " << t.expected << "\n";
        std::cout << "\tActual:   " << actual << "\n";
    }

    std::cout << "\n" << correct << " / " << tests.size() << " correct.\n";
    return 0;
}
